src/rig_reconfigure.cpp

```diff
@@ -76,6 +77,28 @@ int main(int argc, char *argv[]) {
     glfwMakeContextCurrent(window);
     glfwSwapInterval(1); // Enable vsync
 
+    // load window icon
+    auto logoPath = std::filesystem::path(argv[0]).parent_path().append("resources/icon.png");
+
+    std::vector<unsigned char> iconData;
+    unsigned int width;
+    unsigned int height;
+
+    unsigned error = lodepng::decode(iconData, width, height, logoPath.string());
+
+    if (!error) {
+        GLFWimage icon;
+
+        icon.width = static_cast<int>(width);
+        icon.height = static_cast<int>(height);
+        icon.pixels = iconData.data();
+
+        glfwSetWindowIcon(window, 1, &icon);
```

Comment:
Ich lese hier in den docs (https://www.glfw.org/docs/3.3/group__window.html#gadd7ccd39fe7a7d1f0904666ae5932dc5) die remarks

> macOS: The GLFW window has no icon, as it is not a document window, so this function does nothing. The dock icon will be the same as the application bundle's icon. For more information on bundles, see the [Bundle Programming Guide](https://developer.apple.com/library/mac/documentation/CoreFoundation/Conceptual/CFBundles/) in the Mac Developer Library.
> Wayland: There is no existing protocol to change an icon, the window will thus inherit the one defined in the application's desktop file. This function always emits [GLFW_PLATFORM_ERROR](https://www.glfw.org/docs/3.3/group__errors.html#gad44162d78100ea5e87cdd38426b8c7a1).

Vielleicht wäre es sinnvoll das (zusätzlich) noch via desktop-file zu integrieren? Vielleicht könnte man dann sogar starten ohne `ros2 run ...` ermöglichen, bei der gelegenheit vielleicht auch direkt cmd shortcut wie bei uns intern

Reply:
Vielleicht könnte man dann auch via XDG oder so an das icon kommen, egal wo genau das jetzt liegt

Reply:
habe ament zu dem zweck verwendet... desktop file hab ich gebaut, aber nicht getestet... vielleicht hilfts jemandem